src/main/webapp/js/main.js

```diff
@@ -3,11 +3,15 @@ messaging.usePublicVapidKey("BLIwYdJwu8oCxTYQttElp90iDYg5ooy6tzLvx1g2YznQY2JTIQu
 Notification.requestPermission(function(status) {
     console.log('Notification permission status<:', status);
 });
-
+var xhttp = new XMLHttpRequest();
```

Comment:
Optional: if you want to be fancy, you could look at the Fetch API, which is a little cleaner than XHR.